genesyscloud/architect_ivr/genesyscloud_architect_ivr_proxy.go

```diff
@@ -0,0 +1,290 @@
+	return a.deleteArchitectIvrAttr(ctx, a, id)
+}
+
+// createArchitectIvrBasicFn creates a Genesys Cloud Architect IVR (without chunking logic)
+func (a *architectIvrProxy) createArchitectIvrBasic(ctx context.Context, ivr platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.createArchitectIvrBasicAttr(ctx, a, ivr)
+}
+
+// updateArchitectIvrBasic updates a Genesys Cloud Architect IVR (without chunking logic)
+func (a *architectIvrProxy) updateArchitectIvrBasic(ctx context.Context, id string, ivr platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.updateArchitectIvrBasicAttr(ctx, a, id, ivr)
+}
+
+// createArchitectIvrFn is an implementation function for creating a Genesys Cloud Architect IVR
+func createArchitectIvrFn(ctx context.Context, a *architectIvrProxy, ivr platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.uploadArchitectIvrWithChunkingLogic(ctx, true, "", ivr)
+}
+
+// getArchitectIvrFn is an implementation function for retrieving a Genesys Cloud Architect IVR by ID
+func getArchitectIvrFn(_ context.Context, a *architectIvrProxy, id string) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.api.GetArchitectIvr(id)
+}
+
+// updateArchitectIvrFn is an implementation function for updating a Genesys Cloud Architect IVR
+func updateArchitectIvrFn(ctx context.Context, a *architectIvrProxy, id string, ivr platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.uploadArchitectIvrWithChunkingLogic(ctx, false, id, ivr)
+}
+
+// createArchitectIvrBasicFn is an implementation function for performing a basic post of a Genesys Cloud Architect IVR
+// without any chunking logic for the dnis field
+func createArchitectIvrBasicFn(_ context.Context, a *architectIvrProxy, ivr platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.api.PostArchitectIvrs(ivr)
+}
+
+// updateArchitectIvrBasicFn is an implementation function for performing a basic put of a Genesys Cloud Architect IVR
+// without any chunking logic for the dnis field
+func updateArchitectIvrBasicFn(_ context.Context, a *architectIvrProxy, id string, ivr platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return a.api.PutArchitectIvr(id, ivr)
+}
+
+// deleteArchitectIvrFn is an implementation function for deleting a Genesys Cloud Architect IVR
+func deleteArchitectIvrFn(_ context.Context, a *architectIvrProxy, id string) (*platformclientv2.APIResponse, error) {
+	return a.api.DeleteArchitectIvr(id)
+}
+
+// getAllArchitectIvrsFn is an implementation function for retrieving all Genesys Cloud Architect IVRs
+func getAllArchitectIvrsFn(_ context.Context, a *architectIvrProxy, name string) (*[]platformclientv2.Ivr, error) {
+	var allIvrs []platformclientv2.Ivr
+	for pageNum := 1; ; pageNum++ {
```

Comment:
We need to change this pattern.  While almost all APIs return a 200 if you try to call something out of bounds some do not.  We should call the api first, store the first page results and then get the pageCount and call up to the pageCount